### DISCUSSION TITLE
BrowserTabActivity is not found in MAUI

### DISCUSSION THREAD

**Discussion by gazeing** [dotnet/maui#1034] [Q&A]
I'm trying to integrate Azure AD B2C into MAUI. Basically, I was starting with the [dotnet6 mobile sample ](https://github.com/dotnet/net6-mobile-samples) and following the Xamarin instruction [here](https://docs.microsoft.com/en-us/samples/azure-samples/active-directory-b2c-xamarin-native/integrate-azure-ad-b2c-xamarin-forms/) with some tweaks.

I've added Microsoft.Identity.Client library into HelloMaui.csproj file:
```
  <ItemGroup>
    <PackageReference Include="Microsoft.Identity.Client" Version="4.30.1" />
    <PackageReference Include="Microsoft.Maui" Version="6.0.100-preview.3.322" />
    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
  </ItemGroup>
```

And I've tried to extend BrowserTabActivity:

```
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Microsoft.Identity.Client;


namespace HelloMaui
{
    [Activity]
    [IntentFilter(new[] { Intent.ActionView },
        Categories = new[] { Intent.CategoryBrowsable, Intent.CategoryDefault },
        DataHost = "auth",
        DataScheme = "msal90c0fe63-bcf2-44d5-8fb7-b8bbc0b29dc6")]
    public class MsalActivity : BrowserTabActivity
    {
    }
}
```

However, I've got an error when I try to build:
`/net6-mobile-samples/HelloMaui/Android/MsalActivity.cs(17,33): error CS0246: The type or namespace name 'BrowserTabActivity' could not be found (are you missing a using directive or an assembly reference?) [/Users/sxu/work/demo/maui-samples/net6-mobile-samples/HelloMaui/HelloMaui.csproj]`

I thought BrowserTabActivity was a member of Microsoft.Identity.Client library and I can reference the other members of the library like "MsalUiRequiredException" without any problem. Could anyone please let me know what could be the reason for this error? Thanks!

**Comment by rbanks54** [comment]
The .csproj for the identity client only includes the BrowserTabActivity when targeting the MonoAndroid runtimes, as per:

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/0d53f3e80e29f9c6d172e069c10e02d71e24f65e/src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj#L245

I assume we'll need a .NET 6.0./MAUI build of the MSAL library to be able to use it.
I've created an issue at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2655 to see if the MSAL team's can help.

**Comment by pmaytak** [comment]
I was able to repro the issue and found a way that works. I think this is related to Visual Studio versions 2019 Preview vs 2022 Preview. I added MSAL NuGet reference to the [HelloMaui](https://github.com/dotnet/maui-samples/tree/main/HelloMaui) Android project. When I built the project with VS 2019, it pulled netcoreapp2.1 MSAL binary (which doesn't have the Android stuff). When I built with VS 2022, it pulled the correct monoandroid binary and built the project successfully. From the official docs, seems like 2022 should be used for Maui development.

Can you try with VS 2022 to see if it works for you?

I used VS 2019 (16.11.0 Preview 3.0) and VS 2022 (17.0.0 Preview 2.1) for this.

For reference [Net6.0 precedence rules](https://github.com/terrajobst/designs/blob/main/accepted/2021/net6.0-tfms/net6.0-tfms.md#compatibility-rules)

**Reply by rbanks54** [reply]
I'll check it out and see if it works, especially from the command line as that's where I build things from (and because VS2022 on Mac isn't the same as on Windows)

**Reply by rbanks-willow** [reply]
A quick check and the sample code at the top of the thread now compiles. That's a good thing! If anything else comes up I'll raise an issue.

`dotnet build` works from the CLI as well.

Thank you for following up!
